static/resizeiframe.js

```diff
@@ -0,0 +1,25 @@
+document.addEventListener("DOMContentLoaded", function() {
+  var acosAplusResizeIframe = function($, window, document, undefined) {
+    var exerciseWrapper = $('#exercise-page-content'); // A+ adds this around exercise content
+    var newWidth = Math.max(exerciseWrapper.width(), 500); // full width of the exercise area in the A+ page
+    var newHeight = Math.max($(window).height() * 0.8, 500); // 80% of viewport height
+    $('.acos-iframe').width(newWidth).height(newHeight);
+  };
+  
+  var init = function($, window, document) {
+    acosAplusResizeIframe($, window, document);
+    $(window).on('resize', function() {
+      acosAplusResizeIframe($, window, document);
+    });
+  };
+
+  if (typeof require === 'function') {
+    // in Moodle
```

Comment:
We have a Moodle plugin that uses the same A+ protocol, so we wouldn't want to have a separate protocol implementation in acos-server for Moodle. However, Moodle uses AMD modules (require.js) in frontend JavaScript and jQuery should not be defined in the global JS namespace like it is in A+ (A+ currently uses no JS module framework). Therefore, a simple script like `resizeiframe.js` will not work in Moodle without the `require "jquery"` call, while A+ has no `require` function defined at all, hence the if-else branching is needed. If we can't have this simple Moodle-specific logic in this script, it will just crash in Moodle (something like TypeError: jQuery is not defined). I thought that this solution is the most convenient and works for both A+ and Moodle. Another approach would be to use inline JS code in a `<script>` element, but even then it would need a Moodle-specific data attribute so that Moodle can wrap the code into a require call ([README](https://github.com/Aalto-LeTech/mooc-grader/blob/master/doc/README_Moodle_frontend.md#javascript-and-ajax-in-learning-objects) about that).